### DISCUSSION TITLE
the default registry url is a magic value meaning "the currently configured registry".

### DISCUSSION THREAD

**Discussion by everett1992** [npm/feedback#544] [CLI] [answered]
Currently registry.npmjs.org has special meaning when used in package-lock.json resolved field - it is an alias for the currently configured registry. If the behavior is documented, I can't find it. I think other registries should be able to enable this behavior.

https://github.com/npm/arborist/blob/478871bf0a44a8ec516b9057585b8707e60b0349/lib/arborist/reify.js#L687-L693

If your package-lock has`"resolved": "https://registry.npmjs.org/foo/bar/baz"` and your registry is configured `https://private.registry.org` npm will install `https://private.registry.org/foo/bar/baz`. This is a nice feature because it lets you create a package-lock with the default registry then change to different registries without needing to change your package-lock.

However if you modify the package-lock while you have a different registry set the resolved field will use your registry, so if you change to another registry npm will use the old registry for your locked packages. If your package-lock has`"resolved": "https://old.registry.org/foo/bar/baz"` and your registry is configured `https://old.registry.org` npm will install `https://old.registry.org/foo/bar/baz`.

I understand this behavior as well, after all a package's dist.tarball host could be different than it's registry and shouldn't change when the registry changes.

I think the behavior needs to be better documented, and there should be a way to enable the behavior for other registries.

**Comment by MylesBorins** [comment, marked as answer]
@everett1992 if you would like to make documentation changes we accept pull requests at https://github.com/npm/documentation for our general documentation and https://github.com/npm/cli for our CLI speicfic documentatin

**Reply by everett1992** [reply]
I don't have the knowledge to write that pull request. I can make some guesses and start it, but I'll need help to complete it.

**Reply by MylesBorins** [reply]
Open a PR with the best you can do and we'll take it from there 👍

**Reply by everett1992** [reply]
Here's the docs cr: https://github.com/npm/documentation/pull/38
